x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/overall_average.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedIndexPatternColumn, ReferenceBasedIndexPatternColumn } from '../column_types';
+import { optionallHistogramBasedOperationToExpression } from './utils';
+import { OperationDefinition } from '..';
+import { getFormatFromPreviousColumn } from '../helpers';
+
+const ofName = (name?: string) => {
+  return i18n.translate('xpack.lens.indexPattern.overallAverageOf', {
+    defaultMessage: 'Overall average of {name}',
+    values: {
+      name:
+        name ??
+        i18n.translate('xpack.lens.indexPattern.incompleteOperation', {
+          defaultMessage: '(incomplete)',
+        }),
+    },
+  });
+};
+
+export type OverallAverageIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'overall_average';
+  };
+
+export const overallAverageOperation: OperationDefinition<
+  OverallAverageIndexPatternColumn,
+  'fullReference'
+> = {
+  type: 'overall_average',
+  priority: 1,
+  displayName: i18n.translate('xpack.lens.indexPattern.overallAverage', {
+    defaultMessage: 'Overall average',
+  }),
+  input: 'fullReference',
+  selectionStyle: 'hidden',
+  requiredReferences: [
+    {
+      input: ['field', 'managedReference'],
```

Comment:
Seems like it _could_ take a `fullReference` as input, not entirely sure whether it _should_. Thoughts?